Makefile

```diff
@@ -132,28 +133,19 @@ test: test-unit test-e2e
 test-unit:
 	go test -race -short $(PKGS) -count=1
 
+.PHONY: test-rules
+test-rules: $(PROMTOOL_BIN) $(GOJSONTOYAML_BIN) $(ASSETS)
+	@promtool check rules <(cat assets/prometheus-k8s/rules.yaml | $(GOJSONTOYAML_BIN) -yamltojson | jq .spec) | tee tmp/rules.out
+	@grep duplicate tmp/rules.out && exit 1
+
 .PHONY: test-e2e
 test-e2e: KUBECONFIG?=$(HOME)/.kube/config
 test-e2e:
 	go test -v -timeout=20m ./test/e2e/ --kubeconfig $(KUBECONFIG)
 
-.PHONY: test-sec
-test-sec:
-	@which gosec 2> /dev/null >&1 || { echo "gosec must be installed to lint code";  exit 1; }
-	gosec -severity medium --confidence medium -quiet ./...
-
-############
-# Binaries #
-############
-
-$(EMBEDMD_BIN):
-	@go install -mod=vendor github.com/campoy/embedmd
-
-$(GOBINDATA_BIN):
-	@go install -mod=vendor github.com/go-bindata/go-bindata/...
-
-$(JB_BINARY):
-	@go install -mod=vendor github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
+$(BIN_DIR):
+	mkdir -p $(BIN_DIR)
 
-$(GOJSONTOYAML_BINARY):
-	@go install -mod=vendor github.com/brancz/gojsontoyaml
+$(TOOLING): $(BIN_DIR)
+	@echo Installing tools from hack/tools.go
+	@cd hack/tools && cat tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go build -o $(BIN_DIR) %
```

Comment:
`go list -tags tools -f '{{ range .Imports }}{{ printf "%s\n" .}}{{end}}' ./hack/tools` can return the list of imported packages without messing with `grep` and `awk`.

Reply:
I tried that and it doesn't work because:
```
go: directory hack/tools is outside main module
```

Reply:
When I tested, I did `cd hack/tools` indeed :)